src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -0,0 +1,683 @@
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug
+{
+
+    /// <summary>
+    /// Manages the set of Debug profiles and web server settings and provides these as a dataflow source. Note 
+    /// that many of the methods are protected so that unit tests can derive from this class and poke them as
+    /// needed w/o making them public
+    /// </summary>
+    [Export (typeof(ILaunchSettingsProvider))]
+    [AppliesTo(ProjectCapability.LaunchProfiles)]
+    internal class LaunchSettingsProvider : OnceInitializedOnceDisposed, ILaunchSettingsProvider
+    {
+
+        [ImportingConstructor]
+        public LaunchSettingsProvider(UnconfiguredProject unconfiguredProject, IUnconfiguredProjectServices projectServices, 
+                                      IFileSystem fileSystem,   IUnconfiguredProjectCommonServices commonProjectServices, 
+                                      IActiveConfiguredProjectSubscriptionService projectSubscriptionService,
+                                      [Import(AllowDefault = true)]Lazy<ISourceCodeControlIntegration> sourceControlIntegration)
+        {
+            SourceControlIntegration = sourceControlIntegration;
+            ProjectServices = projectServices;
+            FileManager = fileSystem;
+            CommonProjectServices = commonProjectServices;
+            JsonSerializationProviders = new OrderPrecedenceImportCollection<Lazy<ILaunchSettingsSerializationProvider, IJsonSection>>(ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesFirst, 
+                                                                                                                    unconfiguredProject);;
+            ProjectSubscriptionService = projectSubscriptionService;
+        }
+
+        //protected IProjectErrorManager ProjectErrorManager { get; }
+
+        private Lazy<ISourceCodeControlIntegration> SourceControlIntegration { get; }
+        private IUnconfiguredProjectServices ProjectServices { get; }
+        private IUnconfiguredProjectCommonServices CommonProjectServices { get; }
+        private IActiveConfiguredProjectSubscriptionService ProjectSubscriptionService { get; }
+
+        [ImportMany]
+        protected OrderPrecedenceImportCollection<Lazy<ILaunchSettingsSerializationProvider, IJsonSection>> JsonSerializationProviders { get; set; }
+
+        // The source for our dataflow
+        private IReceivableSourceBlock<ILaunchSettings> _changedSourceBlock;
+        protected BroadcastBlock<ILaunchSettings> _broadcastBlock;
+
+        protected IFileSystem FileManager { get; set; }
+
+        // Used to track our errors so we can flush them later
+        public  const string ErrorOwnerString = "LaunchSettingProvider";
+        public  const string IISExpressProfileName = "IIS Express";
```

Comment:
This looks like the only reference to IIS in product code (outside of comments and tests)? We should move this out to the tests as it's used only there.